deps/build.jl

```diff
@@ -1,133 +1,95 @@
+
+    #Linux(:aarch64, :musl)  => ("$bin_prefix/ffmpeg-$v-aarch64-linux-musl.tar.gz", ""),
+    #Linux(:armv7l, :musl)   => ("$bin_prefix/ffmpeg-$v-arm-linux-musleabihf.tar.gz", ""),
+    #Linux(:i686, :musl)     => ("$bin_prefix/ffmpeg-$v-i686-linux-musl.tar.gz", ""),
+    #Linux(:x86_64, :musl)   => ("$bin_prefix/ffmpeg-$v-x86_64-linux-musl.tar.gz", ""),
+
+    #FreeBSD(:x86_64)        => ("$bin_prefix/ffmpeg-$v.x86_64-unknown-freebsd11.1.tar.gz", ""),
+    MacOS(:x86_64)          => ("$bin_prefix/ffmpeg-$v-macos64-shared.tar.gz", "04849d77a027320e53ef2585bcf5b545e094b5c363d39d2b01dfd35ad337c390"),
+
+    Windows(:i686)          => ("$bin_prefix/ffmpeg-$v-win32-shared.tar.gz", "3d6486189953a8f9af14190fdcf43a3216654ee0d64a741b6e108b7d8f7039e5"),
+    Windows(:x86_64)        => ("$bin_prefix/ffmpeg-$v-win64-shared.tar.gz", "19191f33c2ffd437fde8d61cb65724d672389e4522716496829a4a7366a5bf76"),
+)
 
-if Sys.iswindows()
-    provides(Binaries, URI("http://ffmpeg.zeranoe.com/builds/win$(Sys.WORD_SIZE)/shared/ffmpeg-4.1-win$(Sys.WORD_SIZE)-shared.zip"),
-             libav_libs,
-             os = :Windows,
-             unpacked_dir="ffmpeg-4.1-win$(Sys.WORD_SIZE)-shared/bin")
+if Sys.islinux()
+    dependencies = [
+        "https://github.com/SimonDanisch/FribidiBuilder/releases/download/0.14.0/build_fribidi.v0.14.0.jl",
+        # "https://github.com/SimonDanisch/NASMBuilder/releases/download/2.13.3/build_nasm.v2.13.3.jl",
+        "https://github.com/JuliaGraphics/FreeTypeBuilder/releases/download/v2.9.1-3/build_FreeType2.v2.9.1.jl",
+        "https://github.com/JuliaIO/LibassBuilder/releases/download/v0.14.0/build_libass.v0.14.0.jl",
+        "https://github.com/SimonDanisch/FDKBuilder/releases/download/0.1.6/build_libfdk.v0.1.6.jl",
+        "https://github.com/SimonDanisch/LAMEBuilder/releases/download/3.100.0/build_liblame.v3.100.0.jl",
+        "https://github.com/staticfloat/OggBuilder/releases/download/v1.3.3-7/build_Ogg.v1.3.3.jl",
+        "https://github.com/JuliaIO/LibVorbisBuilder/releases/download/v1.3.6/build_libvorbis.v1.3.6.jl",
+        "https://github.com/jpsamaroo/LibVPXBuilder/releases/download/v5.0.0/build_LibVPX.v5.0.0.jl",
+        "https://github.com/jpsamaroo/x264Builder/releases/download/v2018.2.12-noyasm/build_x264Builder.v2018.2.12-pre-noyasm.jl"
+    ]
+
+    for dependency in dependencies
+        file = joinpath(@__DIR__, basename(dependency))
+        isfile(file) || download(dependency, file)
+        # it's a bit faster to run the build in an anonymous module instead of
+        # starting a new julia process
+
+        # Build the dependencies
+        Mod = @eval module Anon end
+        Mod.include(file)
+    end
+    # Not sure why those end up in lib, but they do -.-
+
+    for elem in readdir(joinpath(prefix, "lib"))
+        if occursin(".so", elem)
+            cp(joinpath(prefix, "lib", elem), joinpath(prefix, "bin", elem), follow_symlinks = true, force = true)
+        end
+    end
```

Comment:
I was actually curious why the libraries ended up in `bin`, rather than `lib`... @SimonDanisch, is there a reason why `BinaryProvider` puts everything in `bin`, rather than in `lib`?

Reply:
Actually, that's my doing.. the ffmpeg shared tarballs which are used for Windows and MacOS are from [here](https://ffmpeg.zeranoe.com/builds/), and they bundle everything into `bin` so I modified the `BinaryProvider` tarball to have everything in `bin` too.

Is the current situation ok, or should I edit the tarballs? It would be good to start off on the right foot!

Reply:
I've decided to leave it as-is. Next time we release a new FFMPEG version we can conform to the `bin` and `lib` standard